src/libopenrave-core/jsonparser/jsondownloader.cpp

```diff
@@ -49,21 +72,59 @@ static void _ParseURI(const std::string& uri, std::string& scheme, std::string&
     }
 }
 
-static void _ParseMsgPackDocument(const rapidjson::StringBuffer& buffer, const std::string& uri, rapidjson::Document& doc)
-{
-    try {
-        MsgPack::ParseMsgPack(doc, buffer.GetString(), buffer.GetSize());
+// Need to forward declare this
+static void _ParseDocument(OpenRAVE::JSONDownloadContextPtr pContext);
+
+static bool _ParseEncryptedDocument(OpenRAVE::JSONDownloadContextPtr pContext) {
+#if OPENRAVE_ENCRYPTION
+    // gpgme_check_version must be executed at least once before we create a context.
+    RAVELOG_DEBUG("GPG version: %s\n", gpgme_check_version(NULL));
```

Comment:
RAVELOG_DEBUG shortcuts and we have to ensure gpgme_check_version.

```suggestion
    {
        // gpgme_check_version must be executed at least once before we create a context.
        const char *gpgmeVersion = gpgme_check_version(NULL);
        RAVELOG_DEBUG("GPG version: %s\n", gpgmeVersion);
    }
```

Reply:
I will note this fix, but I will do it in a different way as I am preparing some new code.